src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -134,31 +133,27 @@ private async Task PurgePreviousEventVersions()
 
                     foreach (var table in tables)
                     {
-                        var tableRecords = await table.ExecuteQuerySegmentedAsync(new TableQuery<DiagnosticEvent>(), null);
-
-                        // Skip tables that have 0 records
-                        if (tableRecords.Results.Count == 0)
-                        {
-                            continue;
-                        }
-
-                        // Delete table if it doesn't have records with EventVersion
-                        var eventVersionDoesNotExists = tableRecords.Results.Any(record => string.IsNullOrEmpty(record.EventVersion) == true);
-                        if (eventVersionDoesNotExists)
-                        {
-                            _logger.LogDebug("Deleting table '{tableName}' as it contains records without an EventVersion.", table.Name);
-                            await table.DeleteIfExistsAsync();
-                            tableDeleted = true;
-                            continue;
-                        }
+                        var tableQuery = table.QueryAsync<DiagnosticEvent>(cancellationToken: default);
 
-                        // If the table does have EventVersion, query if it is an outdated version
-                        var eventVersionOutdated = tableRecords.Results.Any(record => string.Compare(DiagnosticEvent.CurrentEventVersion, record.EventVersion, StringComparison.Ordinal) > 0);
-                        if (eventVersionOutdated)
+                        await foreach (var record in tableQuery)
```

Comment:
This makes it so that we have to loop each record which might be overkill here - is there no way to better query the table for the two conditions we need here? 

There is no schema so not every record in the table will be the same in terms of which columns they have

Reply:
We could change the implementation trying to see if we can query the server with both conditions and check what returns to perform the clean up, that looks cleaner indeed.

The reasoning behind this implementation is that I've tried to follow the same logic that we currently have to the extent that the new SDK allowed: 

Previously we would dump the whole table and then run a couple of Any() in memory to check for the condition(s). Now we query the table, letting the SDK paginate internally thanks to the IAsyncEnumerable and we check for both conditions on the same iteration, so in the best case scenario we could even end up pulling less records from the table than before 😊.

Reply:
@liliankasem - this is a net improvement over what exists today. We are only loading 1 page of entries at a time, ending the query when the condition is met. Today we load _all_ entries, then check for the conditions.

Pulling the check into the query would be more optimal, but I am not super concerned due to it still being an improvement.

Reply:
Thanks for the info, this helps :)

Reply:
But that isn't to discount that including in a query condition is an even further improvement. But I believe that goes beyond the scope of this PR (which is to just migrate libraries with functionality parity)

Reply:
Completely agree with both of you. Let's focus on migrating everything to Azure.Data.Tables as soon as possible and then I can revisit this on a future PR applying @liliankasem  great suggestion.